src/ranged.cpp

```diff
@@ -1382,12 +1386,24 @@ static int print_ranged_chance( const player &p, const catacurses::window &w, in
     return line_number;
 }
 
+// Whether character can see the creature for aiming
+static bool char_sees( const Character &ch, const Creature &cr )
+{
+    return ch.sees( cr ) || ch.sees_with_infrared( cr ) || ch.sees_with_specials( cr );
+}
+
+// Whether player character can see the creature for aiming
+static bool pl_sees( const Creature &cr )
+{
+    return char_sees( get_player_character(), cr );
+}
+
 // Handle capping aim level when the player cannot see the target tile or there is nothing to aim at.
 static double calculate_aim_cap( const player &p, const tripoint &target )
 {
     double min_recoil = 0.0;
     const Creature *victim = g->critter_at( target, true );
-    if( victim == nullptr || ( !p.sees( *victim ) && !p.sees_with_infrared( *victim ) ) ) {
+    if( victim == nullptr || !char_sees( p, *victim ) ) {
```

Comment:
Hmm. Actually, how precise are special senses?
I've changed this line assuming they're precise enough to make a difference while aiming, but on the other hand description in 'Look around' menu suggests they do not reveal much beyond general location. If that's the case, this part should be reverted